internal/credentials/spiffe.go

```diff
@@ -73,3 +78,40 @@ func SPIFFEIDFromCert(cert *x509.Certificate) *url.URL {
 	}
 	return spiffeID
 }
+
+type partialParsedSPIFFEBundleMap struct {
+	Bundles map[string]json.RawMessage `json:"trust_domains"`
```

Comment:
Nit: Sorry, didn't notice this earlier. But any specific reason for not naming it as `TrustDomains` to match the JSON field name?